aws_lambda_builders/workflows/python_pip/packager.py

```diff
@@ -596,7 +596,7 @@ def main(self, args, env_vars=None, shim=None):
 class PipRunner(object):
     """Wrapper around pip calls used by chalice."""
 
-    _LINK_IS_DIR_PATTERN = "Processing (.+?)\n  Link is a directory, ignoring download_dir"
+    _LINK_IS_DIR_PATTERNS = ["Processing (.+?)\n  Link is a directory, ignoring download_dir", "Processing (.+?)\n"]
```

Comment:
- You can add some docs on why the pattern is changed.
- Match the second one should always match the first. I believe you have already deduplicated this case, but you can simplify it.
- If we only match "Processing sth.", will we introduce non-dir links?

Reply:
1. I can definitely go ahead and add some docs.
2. It is true for this case, but in the future if the output is to change again, there's no guarantee that the new string will be a subset of the old one. Doing it this way should be more extensible and backwards compatible. It also prevents from having to re-write existing pip test stubs that explicitly write out ```Processing (.+?)\n  Link is a directory, ignoring download_dir```. However, if you think removing the ``` Link is a directory, ignoring download_dir``` pattern is more appropriate, I can certainly go ahead and do that.
3. This was my main concern. Searching through the pip code, though, I think this should be okay as it doesn't appear to log this type of pattern anywhere else. This could always change, but that's the nature of this type of implementation.

Reply:
For 2, You have done a good change on making it a list. You can just keep the second one since the first one is not necessary.

Reply:
Won’t that break customers using an older version of pip?

Reply:
I don't think it would because the matched group will be the same in both cases, and anything matched by ```Processing (.+?)\n  Link is a directory, ignoring download_dir``` will also be matched by ```Processing (.+?)\n```. I just tested this and it seems to be fine. If there is backward compatibility concern, I can also include the old pattern.